azure-proto-core-test/LocationTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿using azure_proto_core;
+using NUnit.Framework;
+using System;
+
+namespace azure_proto_core_test
+{
+    public class LocationTests
+    {
+        [SetUp]
+        public void Setup()
+        {
+        }
+
+        [Test]
+        public void CanCreateLocation()
+        {
+            Location tempLocation = new Location("West US");
+            Assert.AreEqual("West US", tempLocation.Name);
+            
+        }
+
+        [Test]
+        public void EqualsToTrue()
+        {
+            Location location1 = new Location("West US");
+            Location location2 = new Location("West US");
+
+            Assert.IsTrue(location1.Equals(location1));
+            Assert.IsTrue(location1.Equals(location2));
+
+            Assert.IsTrue(location1.Equals("West US"));
+        }
+
+        [Test]
+        public void EqualsToFalse()
+        {
+            Location location1 = new Location("West US");
+            Location location2 = new Location("South US");
+
+            Assert.IsFalse(location1.Equals(location2));
+            Assert.IsFalse(location1.Equals("WestUS"));
```

Comment:
move to a separate test since its testing Location to string comparison vs Location to Location